### PR TITLE
Add numbers

### LINKED ISSUE #7: Calculate the sum of two numbers
### Description
You are tasked with writing a simple Python function that takes two numbers as input and returns their sum.

### Inputs
The function will receive two parameters:
- `num1`: This is the first number (can be an integer or a float).
- `num2`: This is the second number (can also be an integer or a float).

### Outputs
The function should return a single value:
- The sum of `num1` and `num2`. The output should also be of the same type as the inputs (i.e., if both inputs are integers, the output should be an integer; if both are floats, the output should be a float).

### Example
Here’s how the function should behave with some example inputs:

1. **Example 1**
   - **Input**: `num1 = 5`, `num2 = 3`
   - **Output**: `8`

2. **Example 2**
   - **Input**: `num1 = 4.5`, `num2 = 3.5`
   - **Output**: `8.0`

3. **Example 3**
   - **Input**: `num1 = -2`, `num2 = 6`
   - **Output**: `4`

### Function Signature

```
def add_numbers(num1, num2):
    pass          # Your implementation goes here
```

### Instructions
1. Implement the function `add_numbers` following the specifications above.
2. Test your function with various inputs to ensure it works correctly. (unittests)
3. Make sure to handle both integers and floats appropriately.

### PR DESCRIPTION
**Pull request by Bikaze**
implement a solution to #7

### CODE REVIEW

File: solutions/add_numbers.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+
+"""
+A module for adding two numbers.
+
+Module contents:
+    - add_numbers: Adds two numbers and returns their sum.
+
+Created on 2024-12-06
+Author: GitHub Copilot
```
Comment (Bikazemcclent) [inline solutions/add_numbers.py:10]: Why did you use a language model? Can't you do it yourself?

File: solutions/add_numbers.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    Examples:
+        >>> add_numbers(5, 3)
+        8
+        >>> add_numbers(4.5, 3.5)
+        8.0
+        >>> add_numbers(-2, 6)
+        4
+    """
+    assert isinstance(num1, (int, float)), "num1 must be an int or float"
+    assert isinstance(num2, (int, float)), "num2 must be an int or float"
```
Comment (Bikazemcclent) [inline solutions/add_numbers.py:39]: I don't understand what these assertions mean? Can you write a little bit of documentation about them.

File: solutions/tests/test_add_numbers.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - Standard cases: typical numbers
+    - Edge cases: zero, negative numbers
+    - Defensive tests: wrong input types, assertions
+
+Created on 2024-12-06
+Author: GitHub Copilot
+"""
+
+import unittest
+from solutions.add_numbers import add_numbers
```
Comment (Bikazemcclent) [inline solutions/tests/test_add_numbers.py:16]:
```suggestion
from ..add_numbers import add_numbers
```

Why don't you try this neet relative importing style.

**Review by Bikazemcclent — COMMENTED** [review]
You did great. Just check on those few areas.

**Comment by Bikaze** [comment]
Fixed everything

**Review by Bikazemcclent — APPROVED** [review]
Looks great!
